globus_cli/helpers/local_server.py

```diff
@@ -139,7 +139,17 @@ def return_code(self, code):
         self._auth_code_queue.put_nowait(code)
 
     def wait_for_code(self):
-        return self._auth_code_queue.get(block=True)
+        # use a non-blocking get() on the queue so that keyboard interrupts
+        # during the wait will be respected
+        # use a 1 second polling interval, though the exact value doesn't
+        # matter
+        # relevant Python issue discussing this behavior:
+        # https://bugs.python.org/issue1360
+        while True:
+            try:
+                return self._auth_code_queue.get(timeout=1)
+            except Queue.Empty:
+                pass
```

Comment:
I just tested this using block+timeout and the workaround behavior is still in effect. Doing it that way we could just do a simple:
`return self._auth_code_queue.get(block=True, timeout=3600)`

and then future maintainers wouldn't have to figure out what we were up to...